pyproject.toml

```diff
@@ -70,13 +61,138 @@ geopandas = [
 ]
 
 [tool.pytest.ini_options]
-markers = [
-    "web: tests that need network connectivity"
-]
+markers = ["web: tests that need network connectivity"]
 xfail_strict = true
 
 filterwarnings = [
     "ignore:distutils Version classes are deprecated.*:DeprecationWarning:pandas.*",
     "ignore:distutils Version classes are deprecated.*:DeprecationWarning:numpy.*",
-    "ignore:The geopandas.dataset module is deprecated"
+    "ignore:The geopandas.dataset module is deprecated",
+]
+
+[tool.black]
+line-length = 88
+
+[tool.ruff]
+line-length = 88
+select = [
+    # pyflakes
+    "F",
+    # pycodestyle
+    "E",
+    "W",
+    # flake8-2020
+    "YTT",
+    # flake8-bugbear
+    "B",
+    # flake8-quotes
+    "Q",
+    # flake8-debugger
+    "T10",
+    # flake8-gettext
+    "INT",
+    # pylint
+    "PLC",
+    "PLE",
+    "PLR",
+    "PLW",
+    # misc lints
+    "PIE",
+    # flake8-pyi
+    "PYI",
+    # tidy imports
+    "TID",
+    # implicit string concatenation
+    "ISC",
+    # type-checking imports
+    "TCH",
+    # comprehensions
+    "C4",
+    # pygrep-hooks
+    "PGH",
+    # Ruff-specific rules
+    "RUF",
+]
+target-version = "py38"
+ignore = [ # space before : (needed for how black formats slicing)
+    # "E203",  # not yet implemented
```

Comment:
We might be able to remove some of these ignores, but I think that's fine to leave as a follow up, let's get ruff in first.

Reply:
Yeah, I suppose. I just wanted a reasonable starting point so went ahead with whatever pandas adopts.